server/controllers/finance/feeCenter.js

```diff
@@ -0,0 +1,285 @@
+    })
+    .then(function () {
+      return lookupFeeCenter(feeCenterId);
+    })
+    .then(function (feeCenter) {
+      res.status(200).json(feeCenter);
+    })
+    .catch(next)
+    .done();
+}
+
+
+/**
+* Remove a fee center in the database
+*
+* @param {object} req The express request object
+* @param {object} res The express response object
+* @param {object} next The express object to pass the control to the next middleware
+*
+* @example
+* // DELETE /fee_centers : delete a fee center
+* var feeCenters = require('finance/fee_centers');
+* feeCenters.remove(req, res, next);
+*/
+
+function remove (req, res, next) {
+  var feeCenterId = req.params.id;
+  var removeFeeCenterQuery = 'DELETE FROM fee_center WHERE id = ?';
+
+  lookupFeeCenter(feeCenterId)
+    .then(function () {
+      return db.exec(removeFeeCenterQuery, [feeCenterId]);
+    })
+    .then(function () {
+      res.sendStatus(204);
+    })
+    .catch(next)
+    .done();
+}
+
+/**
+* Return a fee center details from the database
+*
+* @param {object} req The express request object
+* @param {object} res The express response object
+* @param {object} next The express object to pass the control to the next middleware
+*
+* @example
+* // GET /fee_centers : returns a fee center detail
+* var feeCenters = require('finance/fee_centers');
+* feeCenters.detail(req, res, next);
+*/
+
+function detail(req, res, next) {
+  'use strict';
+
+  lookupFeeCenter(req.params.id)
+    .then(function (row) {
+      res.status(200).json(row);
+    })
+    .catch(next)
+    .done();
+}
+
+/**
+* Return a fee center instance from the database
+*
+* @param {integer} id of a fee center
+*
+*/
+
+function lookupFeeCenter(id) {
+  'use strict';
+
+  var sql =
+    'SELECT fc.id, fc.label, fc.is_cost, fc.note, fc.is_principal, fc.project_id FROM fee_center AS fc WHERE fc.id = ?';
+
+  return db.exec(sql, id)
+    .then(function (rows) {
+      if (rows.length === 0) {
+        throw new NotFound(`Could not find a fee center with id ${id}`);
+      }
+      return rows[0];
+    });
+}
+
+/**
+* Return a fee cost value by scanning the general ledger
+*
+* @param {object} req The express request object
+* @param {object} res The express response object
+* @param {object} next The express object to pass the control to the next middleware
+*
+* @example
+* // GET /fee_centers/:id/value : returns a value of the fee center
+* var feeCenters = require('finance/fee_centers');
+* feeCenters.getFeeValue(req, res, next);
+*/
+
+function getFeeValue (req, res, next){
+  var sql = null, optionalCondition = '';
+
+
+  lookupFeeCenter(req.params.id)
+    .then(function (){
+      sql =
+        'SELECT ac.id FROM account AS ac WHERE ac.fc_id = ? AND ac.is_title=0';
+
+      return db.exec(sql, req.params.id);
+    })
+    .then(function (rows){
+
+      if (rows.length > 0) {
+        rows = rows.map(function (row) { return row.id;});
+        optionalCondition = ' OR gl.account_id IN (' + rows.join(',') + ')';
```

Comment:
It would be much, much better to use `db.exec()`'s parameters instead of doing string concatenation in JS.